pkg/kubelet/kubelet.go

```diff
@@ -1311,7 +1311,14 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 	kl.statusManager.SetPodStatus(pod, apiPodStatus)
 
 	// Kill pod if it should not be running
-	if errOuter := canRunPod(pod); errOuter != nil || pod.DeletionTimestamp != nil || apiPodStatus.Phase == api.PodFailed {
+	errOuter := canRunPod(pod)
+	if errOuter == nil {
+		rs := kl.runtimeState.networkErrors()
```

Comment:
Not entirely sure what you're saying  :-) .  We now start synchronizing before the network is ready, so that we can see the hostNetwork pods.  But we can't actually start pods that don't use hostNetwork, because the network plug is not ready.